backend/typescript/services/implementations/schedulingService.ts

```diff
@@ -133,23 +147,183 @@ class SchedulingService implements ISchedulingService {
   ): Promise<SchedulingDTO> {
     let newScheduling: Scheduling;
     try {
-      newScheduling = await Scheduling.create({
-        donor_id: scheduling.donorId,
-        categories: scheduling.categories,
-        size: scheduling.size,
-        is_pickup: scheduling.isPickup,
-        pickup_location: scheduling.pickupLocation,
-        day_part: scheduling.dayPart,
-        start_time: scheduling.startTime,
-        end_time: scheduling.endTime,
-        status: scheduling.status,
-        volunteer_needed: scheduling.volunteerNeeded,
-        volunteer_time: scheduling.volunteerTime,
-        frequency: scheduling.frequency,
-        recurring_donation_id: scheduling.recurringDonationId,
-        recurring_donation_end_date: scheduling.recurringDonationEndDate,
-        notes: scheduling.notes,
-      });
+      const schedulingFrequency = scheduling.frequency;
+      if (schedulingFrequency === Frequency.ONE_TIME) {
+        newScheduling = await Scheduling.create({
+          donor_id: scheduling.donorId,
+          categories: scheduling.categories,
+          size: scheduling.size,
+          is_pickup: scheduling.isPickup,
+          pickup_location: scheduling.pickupLocation,
+          day_part: scheduling.dayPart,
+          start_time: scheduling.startTime,
+          end_time: scheduling.endTime,
+          status: scheduling.status,
+          volunteer_needed: scheduling.volunteerNeeded,
+          volunteer_time: scheduling.volunteerTime,
+          frequency: scheduling.frequency,
+          recurring_donation_id: scheduling.recurringDonationId,
+          recurring_donation_end_date: scheduling.recurringDonationEndDate,
+          notes: scheduling.notes,
+        });
+      } else {
+        // get new recurring donation id
+        const dbCurrentSchedules: SchedulingDTO[] = await this.getSchedulings();
+        const recurringDonationIds = dbCurrentSchedules.map((item) => {
+          return item.recurringDonationId;
+        });
+
+        const arrayRecurringDonationIds: number[] = [];
+        for (let i = 0; i < recurringDonationIds.length; i += 1) {
+          if (recurringDonationIds[i] === "null") {
+            arrayRecurringDonationIds.push(0);
+          } else {
+            arrayRecurringDonationIds.push(Number(recurringDonationIds[i]));
+          }
+        }
```

Comment:
```suggestion
       recurringDonationIds.forEach((recurringDonationId, i)=> {
       arrayRecurringDonationIds.push(recurringDonationIds[i] ? Number(recurringDonationid) : 0)
       })
```

Reply:
Will refactor all logic for recurring schedule creation post-launch